.changeset/hot-items-smoke.md

```diff
@@ -0,0 +1,63 @@
+---
+'@backstage/plugin-kubernetes-backend': minor
+---
+
+**BREAKING** Allow refreshing Kubernetes clusters
+
+To allow custom `KubernetesClustersSupplier` instances to refresh the list of clusters
+the `getClusters` method is now called whenever the list of clusters is needed.
+
+Existing `KubernetesClustersSupplier` implementations will need to ensure that `getClusters`
+can be called frequently and likely should return a cached result from `getClusters` instead.
+
+For example, here's a simple example of this in `packages/backend/src/plugins/kubernetes.ts`:
+
+```diff
+-import { KubernetesBuilder } from '@backstage/plugin-kubernetes-backend';
++import {
++  ClusterDetails,
++  KubernetesBuilder,
++  KubernetesClustersSupplier,
++} from '@backstage/plugin-kubernetes-backend';
+ import { Router } from 'express';
+ import { PluginEnvironment } from '../types';
++import { Duration } from 'luxon';
++
++export class CustomClustersSupplier implements KubernetesClustersSupplier {
++  private clusterDetails: ClusterDetails[] = [];
++
++  async retrieveClusters() {
++    this.clusterDetails = []; // fetch from somewhere
++  }
++
++  async getClusters(): Promise<ClusterDetails[]> {
++    return this.clusterDetails;
++  }
++}
+
+ export default async function createPlugin(
+   env: PluginEnvironment,
+ ): Promise<Router> {
+-  const { router } = await KubernetesBuilder.createBuilder({
++  const builder = await KubernetesBuilder.createBuilder({
+     logger: env.logger,
+     config: env.config,
+-  }).build();
++  });
++
++  const clusterSupplier = new CustomClustersSupplier();
```

Comment:
should this example go into the readme perhaps?

Reply:
It's in the docs in `backstage/docs/features/kubernetes/installation.md` now and linked to in `backstage/docs/features/kubernetes/configuration.md` as well.